apps/cookbook/src/app/examples/modal-example/embedded-modal-example/embedded-modal-example.component.ts

```diff
@@ -77,6 +79,26 @@ export class EmbeddedModalExampleComponent implements OnInit {
         setTimeout(() => (this.isLoadingAdditionalContent = false), 2000);
       }
     }
+
+    if (this.alertBeforeClose) {
+      this.modal.canDismiss = () => this.validate();
+    }
+  }
+
+  private validate() {
+    if (this.disableAlertBeforeClose) return true;
```

Comment:
Consider flipping this so it aligns with the config checkbox (i.e. alert if checked):
```suggestion
    if (!this.alertBeforeClose) return true;
```